client/lib/site/index.js

```diff
@@ -61,14 +61,16 @@ Site.prototype.attributes = function( attributes ) {
 };
 
 Site.prototype.set = function( attributes ) {
-	var changed = false;
+	let changed = false;
+	const computedAttributes = Object.assign( {}, attributes, getAttributes( attributes ) );
 
-	for ( var prop in attributes ) {
-		if ( attributes.hasOwnProperty( prop ) && ! isEqual( attributes[ prop ], this[ prop ] ) ) {
-			if ( undefined === attributes[ prop ] ) {
+	// Loop only through the original attributes, use computed values for comparison
```

Comment:
Super minor: I get that you intend "original attributes" to mean "not the `computedAttributes` I created above" but it's slightly misleading to me as the "original attributes" could also be considered `this`, and `attributes` contains the new attributes we're setting on the site.